packages/docs/api/equality-algorithm.md

```diff
@@ -0,0 +1,225 @@
+---
+id: equality-algorithm
+title: Equality Algorithm
+---
+
+The equality algorithm is an extensible and parameterized deep equality checker.
+It is used by all functions that check deep equality inside earl.
+
+## Algorithm
+
+To determine is two values are equal the following checks are executed:
+
+1. If the second value is a [Matcher](/guides/using-matchers.md) and it...
+   1. matches the first value, then the values are considered equal;
+   1. doesn't match the first value, then the values are considered unequal.
+1. Every [custom rules](/advanced/plugin-development.md) is checked.
+   1. If it returns undefined the next rule is checked or the algorithm
+      advances.
+   1. If it returns success, then the values are considered equal.
+   1. If it returns error, then the values are considered unequal.
```

Comment:
I think fallback on false makes it impossible to write rules like: `Always compare arrays using ===`, because if comparison fails then the array is going to be compared deeply anyway.